source/hic_hal/target_config.h

```diff
@@ -55,6 +58,14 @@ typedef struct target_cfg {
     uint8_t erase_reset;            /*!< Reset after performing an erase */
     const sector_info_t* sectors_info; 
     int sector_info_length;
+    
+    /*!< Additional flash and ram regions */
```

Comment:
I'd rather have the extra regions done as structs that pair together start and end. And instead of requiring count fields, treat an entry with start and end both set to 0 as the terminator. You might want to increase the counts so you still have max 3 regions + terminator.

Generally, using 0-value list terminators makes it easier to define lists in code because you don't have to make sure to keep a count value defined somewhere in sync. And in a case like this, it saves a little space since you're always allocating the same number of max extra regions.

Another idea: maybe use "extra" instead of "add"? Just so that you don't have to use an incomplete word.

Example:
```c
typedef struct region_info {
    uint32_t start;
    uint32_t end;
} region_info_t;

typedef struct target_cfg {
    // ... existing fields ...

    region_info_t extra_flash[MAX_EXTRA_FLASH_REGION]; //!< Extra flash regions.
    region_info_t extra_ram[MAX_EXTRA_RAM_REGION]; //!< Extra RAM regions.
} target_cfg_t;
```

Then a target definition would look like this:

```c
const target_cfg_t mytarget = {
        // ... standard stuff ...

        .extra_flash = {
                { 0x10000000, 0x1000FFFF },
        },
        .extra_ram = {
                { 0x30000000, 0x30007FFF },
        },
    };
```

(Btw, the `<` would have to be removed on this comment on line 62. A doxygen comment marker of `!<` indicates that the comment _follows_ the definition. If the comment precedes the definition, only `!` should be used.)